app/controllers/api/v2/report_templates_controller.rb

```diff
@@ -107,12 +107,22 @@ def export
       api :POST, "/report_templates/:id/generate/", N_("Generate a report template")
       param :id, :identifier, :required => true
       param :input_values, Hash, :desc => N_('Hash of input values where key is the name of input, value is the value for this input')
+      param :foreground, :bool, allow_nil: true
+      description <<-DOC
+        The reports are generated by background job. It returns an url to poll for data
+        The <b>foreground</b> parameter is used for to force report to be generated in foreground.
+      DOC
 
       def generate
         @composer = ReportComposer.from_api_params(params)
         if @composer.valid?
-          response = @report_template.render(params: params, template_input_values: @composer.template_input_values)
-          send_data response, :filename => @report_template.suggested_report_name.to_s
+          if params[:foreground]
```

Comment:
Isn't this introducing an api change?

Reply:
It is, @ares asked me to bring the discussion on, if we would want to make the generation on backgroud a default option.

Reply:
I have made new endpoint for asynchronous generations, so we don't break the compatibility and actually leave it up to the user educated decision to use the right one for his usecase :)